turtlebot2_drivers/src/kobuki_node.cpp

```diff
@@ -83,50 +97,124 @@ int main(int argc, char * argv[])
   printf("max_vyaw: %f\n", g_max_vyaw);
 
   parameters.sigslots_namespace = "/kobuki";
-  parameters.device_port = "/dev/kobuki";
   parameters.enable_acceleration_limiter = true;
   g_kobuki = new kobuki::Kobuki();
   g_kobuki->init(parameters);
   g_kobuki->enable();
 
-  auto cmd_vel_sub = node->create_subscription<geometry_msgs::msg::Twist>(
-    "cmd_vel", cmdVelCallback, rmw_qos_profile_default);
-  auto odom_pub = node->create_publisher<nav_msgs::msg::Odometry>("odom", rmw_qos_profile_default);
-
   rclcpp::WallRate loop_rate(20);
 
   auto odom_msg = std::make_shared<nav_msgs::msg::Odometry>();
+  odom_msg->header.frame_id = odom_frame;
+  odom_msg->child_frame_id = base_link_frame;
+  auto imu_msg = std::make_shared<sensor_msgs::msg::Imu>();
+  imu_msg->header.frame_id = gyro_link_frame;
+  auto tf_msg = std::make_shared<geometry_msgs::msg::TransformStamped>();
+  tf_msg->header.frame_id = odom_frame;
+  tf_msg->child_frame_id = base_link_frame;
   ecl::LegacyPose2D<double> pose;
 
   while (rclcpp::ok()) {
+    rcl_time_point_value_t now;
+    double gyro_yaw, gyro_vyaw;
     ecl::LegacyPose2D<double> pose_update;
     ecl::linear_algebra::Vector3d pose_update_rates;
     {
       std::lock_guard<std::mutex> kobuki_guard(g_kobuki_mutex);
       g_kobuki->updateOdometry(pose_update, pose_update_rates);
-      auto now = std::chrono::system_clock::now(); 
+      gyro_yaw = g_kobuki->getHeading();
+      gyro_vyaw = g_kobuki->getAngularVelocity();
+      if (rcl_system_time_now(&now) != RCL_RET_OK) {
+        std::cerr << "Failed to get system time" << std::endl;
+      }
+      auto now = std::chrono::system_clock::now();
```

Comment:
Can you change this variable, or the other one? It looks like you have `now` shadowed as both a `rcl_time_point_value_t` and something like a `std::chrono::time_point` (whatever `auto` becomes).